### PR TITLE
YALB-1478: Views: display modes and dynamic lists

### PR DESCRIPTION
**Pull request by codechefmarc**
### Description of work
- Includes work from PR-363 YALB-85/YALB-1496, profile and layout sections.
- Adds condensed display view mode for all content types
- Adds directory display view mode for profiles
- Adds new 1:1 crop, image style, and responsive image style for profile photos
- Adds additional name fields (First, last, and prefix)
- Adds auto entity label module to auto populate the "Display Name" field from "Prefix First Last" if "Display Name" is blank
- Adds directory view block, sorted on "Pin to beginning of list" and then "Last name" ascending
- Views tool now supports "Pin to beginning of list" as the initial sort before the selected sort
- Views tool profile sort is based on "Last name" now instead of "Title" field.

### Todo for Front-End

- Profile Lists - images need to be smaller
- Additional image sizes / responsive image styles for 1:1 profile images?

### Functional testing steps:
- [x] Add 3 profiles to the site, each with different last names to later verify sorting.
- [x] Note: The profile tabs do not show correctly, additional styling is needed and is already noted in ticket [YALB-1501](https://yaleits.atlassian.net/browse/YALB-1501) Therefore, you will have to hover on the top area to see "General Info" and "Contact Info" tabs.

Screenshot 2023-08-07 at 5 36 55 PM

- [x] Verify that there are now 4 "name" fields - "Prefix", "First Name", "Last Name", and "Display Name".

Screenshot 2023-08-07 at 4 56 39 PM

- [x] On one of the profiles, leave the "Display Name" field empty.
- [x] Make sure to also add Media
- [x] When adding new media, verify that in the "Crop image" fieldset, there is a new crop option for "Profile 1:1"
- [x] Visit the "Contact Info" tab (currently hard to find b/c styling) at the top of the form and fill out dummy email, phone, and address.

Screenshot 2023-08-07 at 5 35 37 PM

- [x] After saving the profiles, verify that the one that didn't get a "Display Name" should be automatically filled in with information from the "Prefix", "First Name", and "Last Name" fields.
- [x] Create a page on the site, title it, save, and enter layout builder
- [x] Add a block of type "Profile Directory"
- [x] Verify that the Profile Directory shows a new view for each profile, has the correct 1:1 aspect ratio on images, and that it is sorted by last name.
- [x] Add a block of type "View" and set the view to show "Profiles" as "Grid"
- [x] Verify that the view shows a different view mode for the profiles.
- [x] Change to "List" and to "Condensed" and verify that the new view modes work as intended.
- [x] Change the view to sort on last name descending and verify that the view is re-ordered.
- [x] Edit a profile and, via the sidebar under Publishing settings", check "Pin to the beginning of list". Save the profile.

Screenshot 2023-08-08 at 3 04 27 PM

- [x] Refresh the view and verify that the "pinned" profiles always are sorted first in the list.
- [x] Change the view to show pages, posts, or events. And set the view mode to "Condensed"
- [x] Verify that the condensed view works with pages, posts, and events.

Screenshot 2023-08-08 at 3 06 43 PM

[YALB-1501]: https://yaleits.atlassian.net/browse/YALB-1501?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dblanken-yale** [comment]
@codechefmarc Just verifying that it's ok for the profile crop to not be pre-selected when expanding by default.  Makes sense to me to click profile since I'm in one, but wanted to make sure that wasn't supposed to be captured in this ticket.

Also not sure if captured in this ticket, but I have been used to using the buttons under "Manage X" to add data, however Profiles does not have a button for "Add Profile".  Easily gotten around by visiting the Add Content dropdown, but wanted to document it here in case it should be included here as well or if that's a separate ticket.

Looking great so far!  Still going through testing.

**Review by dblanken-yale — COMMENTED** [review]
I noticed that "Pin to the beginning of the list" shows when creating Events and the like also.  Is this by design?  Will all content types be able to be pinned to the front of views as well?  Neat idea IMO.

Great work!  When those front end changes get through, I think everyone is going to love it.

**Comment by joetower** [comment]
@codechefmarc I pushed a fix for the horizontal tabs. 

https://github.com/yalesites-org/yalesites-project/assets/366413/f516eded-7527-4893-9e4b-398fda4653ac

Also, the `Position` info  `i` icon button should get fixed with the 1501 ticket.

**Comment by codechefmarc** [comment]
@joetower - tabs look great!

@dblanken-yale - Manage Profiles now has an "Add profile" link on the top, thanks for noticing!

As for "Pin to top of list" - yup that should be for any content type and (now) works with the view tool, something I wanted to add a long time ago.

To the crop question, since we're using the same media library media type for all content, the crop settings are showing what the crops can look for any placement. I could move the profile up, but, I didn't want to mess with what people are used to and I think, in general, they will be changing crop for 16:9 and 3:2 more often.

File: web/profiles/custom/yalesites_profile/config/sync/core.entity_view_display.node.page.default.yml [hunk 1]
```diff
@@ -56,6 +56,7 @@ third_party_settings:
         third_party_settings:
           layout_builder_lock:
             lock:
+              1: 1
```
[the remark this answers is not in the document]
Comment (joetower) [reply web/profiles/custom/yalesites_profile/config/sync/core.entity_view_display.node.page.default.yml]: I removed it again.

**Comment by dblanken-yale** [comment]
The Display Name when editing seems to have different styling than the rest of the form.  Is that by design?

Screenshot 2023-08-17 at 2 39 44 PM

**Review by dblanken-yale — APPROVED** [review]
All good now I think.  Great work

**Pull request merged**